src/main/java/edu/cornell/kfs/coa/document/validation/impl/SubAccountGlobalRule.java

```diff
@@ -444,13 +457,369 @@ protected boolean checkCgIcrRules(String subAccountTypeCode, SubAccountGlobalDet
         return success;
     }  
     
+    protected boolean checkNewSubAccountRules() {
+        SubAccountGlobal subAccountGlobal = (SubAccountGlobal) super.getNewBo();
+        
+        if (newSubAccountSectionHasEmptyValues(subAccountGlobal)) {
+            return true;
+        }
+        
+        boolean newAccountFieldsAreEntered = checkAppropriateNewAccountFieldsAreEnteredBasedOnApplyAllCheckboxStatus(subAccountGlobal);
+        boolean accountsExist = checkAccountsExist(subAccountGlobal);
+        boolean success = newAccountFieldsAreEntered && accountsExist;
+        success &= checkBasicGlobalFieldsAndEditSubAccountSectionAreEmpty(subAccountGlobal);
+        success &= checkSubAccountTypeIsSpecified(subAccountGlobal);
```

Comment:
Yes, I think it's safe to run them. One checks to make sure the appropriate sub-account-edit fields are empty, and the other checks to make sure the new sub-account-type field is filled in. This will allow for highlighting more form errors to the user at once. (Below these lines are "if" statements to appropriately skip over other checks that would run into problems if the previous checks failed.)

Reply:
sounds good

Reply:
I thought we decided to change this such that checkBasicGlobalFieldsAndEditSubAccountSectionAreEmpty and checkSubAccountTypeIsSpecified run even if there were previous validation errors

Reply:
The "checkBasicGlobalFieldsAndEditSubAccountSectionAreEmpty" and "checkSubAccountTypeIsSpecified" checks should already be getting run in the current code, even when previous validations fail. For the "newSubAccountSectionHasEmptyValues" check above this code, that's just a short-circuit to skip over this whole validation portion if the new-sub-account fields are not filled in, since we don't need the new-sub-account validation if that section is empty.